lib/schema/generateSchema.js

```diff
@@ -4,46 +4,53 @@ var fs = require('fs'),
 
 exports.generateSchema = function generateSchema(next) {
 	var self = this;
-	if (self.config.schemaRefresh === false) {
-		self.logger.debug('skipping generation of schema, config.schemaRefresh is false');
-		return next && next();
-	}
-
-	/*
-	 Use a hashed filename based on the configured authentication so if
-	 they change their config, we will update the schema too.
-	 */
 	var confDir = self.getCacheDir(),
 		hashedFilename = 'salesforce-schema-' + self.md5((self.config.generateModels || []).join('.') + self.config.username + self.config.token) + '.json';
 
-	// If schemaCache is false, we look it up every time.
-	if (self.config.schemaCache === false) {
-		self.cleanGeneratedModels();
-		return self.describeGlobal(confDir, hashedFilename, next);
-	}
-
 	var cachedFileName = path.join(confDir, hashedFilename),
 		schemaRefresh = self.config.schemaRefresh === undefined ? 3.6e+6 : self.config.schemaRefresh;
```

Comment:
Lines 7, 8, and 10 are part of a single concern. Why don't we put them in a dedicated function that do one thing? We can reuse it when we need.